pallets/runtime/tests/src/relayer_test.rs

```diff
@@ -72,22 +157,28 @@ fn do_basic_relayer_paying_key_test() {
         10_000u128
     ));
 
+    // Bob now has a subsidy of 10,000 POLYX.
+    assert_subsidy(bob, Some((alice, 10_000u128)));
+
     // Dave tries to remove the paying key from Bob's key.  Not allowed.
     TestStorage::set_current_identity(&dave.did);
     assert_noop!(
-        Relayer::remove_paying_key(dave.origin(), bob.acc(), alice.acc(),),
+        Relayer::remove_paying_key(dave.origin(), bob.acc(), alice.acc()),
         Error::NotAuthorizedForUserKey
     );
 
-    // add authorization for using Dave as the paying key for bob
+    // Dave tries to remove the wrong paying key from Bob's key.  Not allowed.
     TestStorage::set_current_identity(&dave.did);
-    assert_ok!(Relayer::set_paying_key(
-        dave.origin(),
-        bob.acc(),
-        0u128.into()
-    ));
+    assert_noop!(
+        Relayer::remove_paying_key(dave.origin(), bob.acc(), dave.acc()),
+        Error::NotPayingKey
+    );
+
+    // Add authorization for using Dave as the paying key for Bob.
+    assert_ok!(Relayer::set_paying_key(dave.origin(), bob.acc(), 0u128));
 
-    // Bob tries to accept the new paying key, but he already has a paying key
+    // Bob tries to accept the new paying key, but he already has a paying key.
+    // TODO: Need to allow changing the paying key.
```

Comment:
Let's create a task for this to fix after this PR?

Reply:
Already done in PR #1057.